bson/src/main/org/bson/RawBsonDocument.java

```diff
@@ -105,6 +106,7 @@ public RawBsonDocument(final byte[] bytes, final int offset, final int length) {
         this.bytes = bytes;
         this.offset = offset;
         this.length = length;
+        this.codecRegistry = createCodecRegistry();
```

Comment:
Thinking of a use case like a backup system that might be backing up millions of small documents using RawBsonDocument for efficiency.  Now each instance has to pay the cost of construction of this registry as well as the extra memory, which with small documents will dwarf the size of the byte array.  It's just too heave a price to pay for the simplicity it buys.   We can save on the cost by sticking with the static registry and instead use conditionals to special case BsonType.DOCUMENT, and eventually BsonType.ARRAY.